src/r2b2/audit.py

```diff
@@ -100,17 +102,53 @@ def current_dist_null(self, kmin: int):
         else:
             round_draw = self.rounds[-1] - self.rounds[-2]
 
+        # Distribution updating is dependent on sampling with or without replacement
         if self.replacement:
             distribution_round_draw = binom.pmf(range(0, round_draw + 1),
                                                 round_draw, 0.5)
-            if len(self.rounds) < 2:
+            # Compute convolution to get new distribution (except 1st round)
+            if len(self.rounds) == 1:
                 self.distribution_null = distribution_round_draw
             else:
                 self.distribution_null = fftconvolve(self.distribution_null,
                                                      distribution_round_draw)
         else:
-            # TODO Implement updating distributions for without replacement
-            return
+            half_contest_ballots = math.floor(self.contest.contest_ballots / 2)
+            if len(self.rounds) == 1:
+                # Simply compute hypergeometric for 1st round distribution
+                self.distribution_null = [
+                    hypergeom.pmf(k, self.contest.contest_ballots,
+                                  half_contest_ballots, round_draw)
+                    for k in range(round_draw + 1)
+                ]
```

Comment:
I think performance will be improved considerably by changing this to self.distribution_null = hypergeom.pmf(range(round_draw + 1), contest_ballots, half_contest_ballots, round_draw)). (Tested on round size of 50000.)